pkg/steps/artifacts_test.go

```diff
@@ -712,3 +717,92 @@ func TestAddPodUtils(t *testing.T) {
 	}
 	testhelper.CompareWithFixture(t, base)
 }
+
+func aPod() *coreapi.Pod {
+	return &coreapi.Pod{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      "p",
+			Namespace: "ns",
+		},
+	}
+}
+
+func TestWaitForConditionOnObject(t *testing.T) {
+	containerName := "c"
+	podName := "p"
+	ns := "ns"
+
+	evaluateFunc := func(obj runtime.Object) (bool, error) {
+		switch pod := obj.(type) {
+		case *coreapi.Pod:
+			for _, container := range pod.Status.ContainerStatuses {
+				if container.Name == containerName {
+					if container.State.Running != nil || container.State.Terminated != nil {
+						return true, nil
+					}
+					break
+				}
+			}
+		default:
+			return false, fmt.Errorf("pod/%v ns/%v got an event that did not contain a pod: %v", podName, ns, obj)
+		}
+		return false, nil
+	}
+
+	testCases := []struct {
+		name          string
+		expected      error
+		client        ctrlruntimeclient.WithWatch
+		containerName string
+		objectFunc    func(client ctrlruntimeclient.Client, t *testing.T)
+	}{
+		{
+			name:   "happy path: pod",
+			client: fakectrlruntimeclient.NewFakeClient(aPod()),
+			objectFunc: func(client ctrlruntimeclient.Client, t *testing.T) {
+				if err := wait.Poll(100*time.Millisecond, time.Second,
+					func() (bool, error) {
+						ctx := context.TODO()
+						pod := &coreapi.Pod{}
+						if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Name: podName, Namespace: ns}, pod); err != nil {
```

Comment:
What creates this Pod?

Reply:
It comes with the fake client.

https://github.com/openshift/ci-tools/pull/1874/files#diff-86a943161d4f6ebe47a0c7efd74c4ac67371f1f4e99d4a32c8ea56e4d859f1ccR761